src/dso_api/dynamic_api/filters/values.py

```diff
@@ -0,0 +1,136 @@
+"""Conversion of query-string values to Python types."""
+from __future__ import annotations
+
+import math
+import re
+from datetime import date, datetime, time
+from decimal import Decimal
+
+from django.contrib.gis.geos import GEOSException, GEOSGeometry, Point
+from django.utils.dateparse import parse_datetime
+from django.utils.translation import gettext_lazy as _
+from gisserver.geometries import CRS
+from rest_framework.exceptions import ValidationError
+
+# Don't want Decimal("NaN"), Decimal("-inf") or '0.321000e+2' to be accepted.
+RE_DECIMAL = re.compile(r"^[0-9]+(\.[0-9]+)?$")
+
+
+def str2bool(value: str) -> bool:
+    value = str(value).lower()
+    if value in ("true", "1"):
+        return True
+    elif value in ("false", "0"):
+        return False
+    else:
+        raise ValueError("expect true/false")
+
+
+def str2number(value: str) -> Decimal:
+    if not RE_DECIMAL.match(value):
+        raise ValueError("expecting number")
+
+    return Decimal(value)
+
+
+def str2isodate(value: str) -> date | datetime | None:
+    """Parse ISO date and datetime."""
+
+    # Try parsing full iso-8601 datetime.
+    result = parse_datetime(value)
+    if result is not None:
+        return result
```

Comment:
`parse_datetime` can raise `ValueError` "if the input is well formatted but not a valid datetime". Is that accounted for?

Reply:
Yes, but my point is that `parse_datetime` is not inside a `ValueError` block but the code below is...

Reply:
ok, fixed!